packages/realm/src/Realm.ts

```diff
@@ -1237,6 +1242,13 @@ export class Realm {
     return this.classes.getHelpers<T>(arg);
   }
 
+  /**
+   * @internal
+   */
+  private updateClassMap() {
+    this.classes = new ClassMap(this, this.internal.schema, this.schema);
```

Comment:
For a large schema, it is an expensive operation to change even small things. Would be possible to apply changes to the class map and not start all over each time? Maybe not in this PR but creating an issue to track it?

Reply:
Nice observation! It'd definitely be interesting to look into what we could do to optimize this. I'll create a ticket.